graspologic/partition/leiden.py

```diff
@@ -412,15 +425,15 @@ def hierarchical_leiden(
     max_cluster_size: int = 1000,
     starting_communities: Optional[Dict[str, int]] = None,
     extra_forced_iterations: int = 0,
-    resolution: float = 1.0,
-    randomness: float = 0.001,
+    resolution: Union[int, float] = 1.0,
```

Comment:
FYI that `float` is treated as `Union[float, int]` per the PEP-whatever on type hinting

Reply:
beartype wasn't happy though

Reply:
I was just thinking "I need a `FloatLike` or something defined somewhere".  `Floatational` even. :fingerguns: